src/cpp/types/endpoint/Endpoint.cpp

```diff
@@ -107,8 +108,30 @@ std::ostream& operator <<(
         std::ostream& os,
         const Endpoint& endpoint)
 {
+    std::string kind_str;
+    switch (endpoint.kind_)
```

Comment:
This switch is better done for its proper object:
`std::ostream& operator <<(std::ostream& os, const EndpointKind& endpoint)`